### PR TITLE
[RFR] Fix Datagrid can't handle dynamic children

### LINKED ISSUE #3631: [alpha.3] Dynamical DataGrid column changing not followed with List rerendering
**What you were expecting:**
I expect to be able to dynamically configure columns set in DataGrid

**What happened instead:**
Only header row changes its configuration, on column config change without page reload.

**Steps to reproduce:**

- Go to codesandbox
- Open Posts view
- Click first  `Toggle Columns` button
- Consider only header row config changed

**Related code:**
https://codesandbox.io/s/simple-ju5rm

Consider `Toggle Columns` button behavior - when columns configuration depends on state, when config in state changes - only header row updates configuration. 
If config stored in the cache - columns are rendered correctly on page reloading.

**Other information:**
This reproduced only on 3.0 alpha version.
Quite similar behavior observed with `ra-customizable-datagrid` module, even if this module rerenders all selected DataGrid children under the hood, instead of conditional rendering one-by-one. Previously it worked fine.
If there are some backward incompatible changes in DataGrid, I'll be glad for some guidance about them, to find a way to fix external package.

**Environment**

* React-admin version: "^3.0.0-alpha.3",
* Last version that did not exhibit the issue (if applicable): 2.9.x
* React version: ~16.8.0
* Browser: Chrome

### PR DESCRIPTION
**Pull request by djhi**
Fixes #3631 and #3629

### CODE REVIEW

**Comment by natrim** [comment]
confirming it fixes #3629

**Comment by djhi** [comment]
@fzaninotto I just tried this branch again with a CPU throttle set to x6 and I haven't notice any performance issues nor differences with the version without this fix. Can you try again ?

File: packages/ra-ui-materialui/src/list/DatagridBody.js [hunk 1]
```diff
@@ -91,7 +82,11 @@ DatagridBody.defaultProps = {
     row: <DatagridRow />,
 };
 
+const MemoDatagridBody = memo(DatagridBody);
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/DatagridBody.js]: memo doesn't work here as `children` will always change

**Comment by fzaninotto** [comment]
I think the default export should be DatagridBody, not MemoDatagridBody

**Pull request merged**